pycbc/results/pygrb_plotting_utils.py

```diff
@@ -0,0 +1,640 @@
+
+    parser.add_argument("-l", "--veto-directory", action="store", default=None,
+                        help="The location of the CATX veto files")
+
+    parser.add_argument("-b", "--veto-category", action="store", type=int,
+                        default=None, help="Apply vetoes up to this level " +
+                        "inclusive")
+
+    parser.add_argument("-i", "--ifo", default=None, help="IFO used for IFO " +
+                        "specific plots")
+
+    parser.add_argument("--use-sngl-ifo-snr", default=False,
+                        action="store_true", help="Plots are vs single IFO " +
+                        "SNR, rather than coherent SNR")
+
+    parser.add_argument("--variable", default=None, help="Quantity to plot " +
+                        "the vertical axis. Supported choices are: " +
+                        "coherent, single, reweighted, or null (for " +
+                        "timeeries plots), standard, bank, or auto (for " +
+                        "chi-square veto plots), coincident, nullstat, " +
+                        "or overwhitened (for null statistics plots)")
+
+    parser.add_argument('--plot-title',
+                        help="If given, use this as the plot caption")
+
+    parser.add_argument('--plot-caption',
+                        help="If given, use this as the plot caption")
+
+    return parser.parse_args()
+
+
+# =============================================================================
+# Format single detector chi-square data as numpy array and floor at 0.005
+# =============================================================================
+
+def format_single_chisqs(trig_ifo_cs, ifos):
+    """Format single IFO chi-square data as numpy array and floor at 0.005"""
+    for ifo in ifos:
+        trig_ifo_cs[ifo] = numpy.asarray(trig_ifo_cs[ifo])
+        numpy.putmask(trig_ifo_cs[ifo], trig_ifo_cs[ifo] == 0, 0.005)
+
+    return trig_ifo_cs
+
+
+# =============================================================================
+# Reset times so that t=0 is corresponds to the GRB trigger time
+# =============================================================================
+
+def reset_times(seg_dir, trig_data, inj_data, inj_file):
+    """Reset times so that t=0 is corresponds to the GRB trigger time"""
+    segs = readSegFiles(seg_dir)
+    grb_time = segs['on'][1] - 1
+    start = int(min(trig_data.time)) - grb_time
+    end = int(max(trig_data.time)) - grb_time
+    duration = end-start
+    start -= duration*0.05
+    end += duration*0.05
+    trig_data.time = [t-grb_time for t in trig_data.time]
+
+    if inj_file:
+        inj_data.time = [t-grb_time for t in inj_data.time]
+
+    return grb_time, start, end, trig_data, inj_data
+
+
+# =============================================================================
+# Extract trigger/injection data produced by PyGRB
+# =============================================================================
+
+class PygrbFilterOutput(object):
```

Comment:
One thing to consider here (not for this patch!) is to use some of python's class functionality to only load data when needed. For e.g. you don't need to read/calculate `rel_amp_1` in most cases. Using something like https://www.geeksforgeeks.org/getter-and-setter-in-python/, which we use also in the all-sky code, you can only calculate these things once and store it in memory if needed after the first time.

Reply:
... This would need the HDF input files anyway